test/extensions/filters/listener/common/fuzz/listener_filter_fuzzer.cc

```diff
@@ -10,19 +8,87 @@ void ListenerFilterFuzzer::fuzz(
     Network::ListenerFilter& filter,
     const test::extensions::filters::listener::FilterFuzzTestCase& input) {
   try {
-    fuzzerSetup(input);
+    socket_.setLocalAddress(Network::Utility::resolveUrl(input.sock().local_address()));
+  } catch (const EnvoyException& e) {
+    // Socket's local address will be nullptr by default if fuzzed local address is malformed
+    // or missing - local address field in proto is optional
+  }
+  try {
+    socket_.setRemoteAddress(Network::Utility::resolveUrl(input.sock().remote_address()));
   } catch (const EnvoyException& e) {
-    ENVOY_LOG_MISC(debug, "EnvoyException: {}", e.what());
-    return;
+    // Socket's remote address will be nullptr by default if fuzzed remote address is malformed
+    // or missing - remote address field in proto is optional
+  }
+
+  const int nreads = input.data_size(); // Number of reads
+
+  if (nreads > 0) {
+    EXPECT_CALL(socket_, detectedTransportProtocol()).WillRepeatedly(testing::Return("raw_buffer"));
+
+    EXPECT_CALL(os_sys_calls_, recv(FAKE_SOCKET_FD, _, _, MSG_PEEK))
+        .WillOnce(testing::Return(Api::SysCallSizeResult{static_cast<ssize_t>(0), 0}));
+
+    EXPECT_CALL(dispatcher_,
+                createFileEvent_(_, _, Event::FileTriggerType::Edge,
+                                 Event::FileReadyType::Read | Event::FileReadyType::Closed))
+        .WillOnce(testing::DoAll(testing::SaveArg<1>(&file_event_callback_),
+                                 testing::ReturnNew<NiceMock<Event::MockFileEvent>>()));
   }
 
   filter.onAccept(cb_);
-}
 
-void ListenerFilterFuzzer::socketSetup(
-    const test::extensions::filters::listener::FilterFuzzTestCase& input) {
-  socket_.setLocalAddress(Network::Utility::resolveUrl(input.sock().local_address()));
-  socket_.setRemoteAddress(Network::Utility::resolveUrl(input.sock().remote_address()));
+  if (nreads > 0) {
+    // Construct header from single or multiple reads
+    std::string data = "";
+    std::vector<size_t> indices; // Ending indices for each read
+
+    size_t curr = 0;
+
+    for (int i = 0; i < nreads; i++) {
+      data += input.data(i);
+      curr += input.data(i).size();
+      indices.push_back(curr);
```

Comment:
Isn't `curr` always just `data.size()`?

Reply:
`data.size()` only returns the size of a particular read, so it might be something like [5, 4, 9]. `curr` should return the sum of all reads so far, so it'd be [5, 9, 18].

Reply:
Sorry, I meant the concatenation of all the `input.data` values, which is a string that is also called `data`. I think `curr` should always just be the length of that concatenation.

Reply:
Yup, that should work! Fixed in the latest commit.